File: src/main/java/myblog/blog/infra/log/TraceId.java
package myblog.blog.infra.log;

import lombok.Getter;

import java.util.UUID;

@Getter
public class TraceId {
    private final String id;
    private final int level;

    public TraceId(String id) {
        this.id = id +"/"+ createdTransactionId();
        this.level = 0;
    }

    private TraceId(String id, int level) {
        this.id = id;
        this.level = level;
    }

    private String createdTransactionId() {
        return UUID.randomUUID().toString().substring(0,8);
    }

    public TraceId createNextId(){
        return new TraceId(id, level+1);
    }

    public TraceId createPrevId(){
        return new TraceId(id, level-1);
    }

    public boolean isFirstLevel(){
        return level == 0;
    }
}

File: libs/global-logging/src/main/kotlin/kr/co/jiniaslog/shared/log/TraceId.kt
package kr.co.jiniaslog.shared.log

import java.util.UUID

data class TraceId(
    val id: String,
    val level: Int,
) {
    fun createNextId(): TraceId {
        return TraceId(id, level + 1)
    }

    fun createPrevId(): TraceId {
        return TraceId(id, level - 1)
    }

    val isFirstLevel: Boolean
        get() = level == 0

    companion object {
        fun create(id: String?): TraceId {
            return TraceId(id + "/" + UUID.randomUUID().toString().substring(0, 8), 0)
        }
    }
}
